### PR TITLE
[server] Move specialised funtions in correponding service cpp files

### PR DESCRIPTION
**Pull request by pblottiere**
## Description

The underlying idea is to simplify the WMS renderer by separating responsibilities. For example, every function/method dedicated to the `GetLegendGraphics` request should be moved in `qgswmsgetlegendgraphics.cpp` instead of being within the renderer itself.

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

- [ ] Commit messages are descriptive and explain the rationale for changes
- [ ] Commits which fix bugs include `fixes #11111` in the commit message next to the description
- [ ] Commits which add new features are tagged with `[FEATURE]` in the commit message
- [ ] Commits which change the UI or existing user workflows are tagged with `[needs-docs]` in the commit message and contain sufficient information in the commit message to be documented
- [ ] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [ ] New unit tests have been added for core changes
- [ ] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTING.md#contributing-to-qgis) before each commit

### CODE REVIEW

File: src/server/services/wms/qgswmsrenderer.h [hunk 1]
```diff
@@ -218,7 +220,7 @@ namespace QgsWms
       /**
        * Checks WIDTH/HEIGHT values against MaxWidth and MaxHeight
         \returns true if width/height values are okay*/
```
Comment (elpaso) [inline src/server/services/wms/qgswmsrenderer.h]: Need to fix this if it now returns void

**Review by elpaso — APPROVED** [review]
I like the idea, even if I'm the middle of some fixes in this area of the code and I'll become crazy rebasing :) 

You beat me on time.

File: src/server/services/wms/qgswmsrenderer.h (hunk 1 above)
Comment (pblottiere) [reply src/server/services/wms/qgswmsrenderer.h]: Indeed, I forgot to update the documentation for now (also for other methods too).

**Comment by pblottiere** [comment]
> I like the idea, even if I'm the middle of some fixes in this area of the code and I'll become crazy rebasing :)

I can totally wait until you are ready if you want?

**Comment by elpaso** [comment]
@pblottiere thanks, that would be awesome!
 
I think I will make the PR tomorrow. It's a bugfix for https://issues.qgis.org/issues/21697 It would have been ready today but I've got sidetracked by the ugliest work of my life (XML validation for INSPIRE metadata).

**Comment by pblottiere** [comment]
> thanks, that would be awesome!

:+1:

**Comment by pblottiere** [comment]
> It would have been ready today but I've got sidetracked by the ugliest work of my life (XML validation for INSPIRE metadata).

Ouch, I wish you *bon courage*!

**Comment by elpaso** [comment]
@pblottiere when https://github.com/qgis/QGIS/pull/9704 is merged you're good to go!

**Pull request merged**